ui-admin/src/portal/publish/diffComponents.tsx

```diff
@@ -67,15 +68,78 @@ export const versionDisplay = (stableId: string, version: number) => {
 }
 
 /** Summary of notification config changes -- doesn't show any detail yet */
-export const ConfigChangeListView = ({ configChangeList }:
-                                      {configChangeList: ListChange<object, VersionedConfigChange>}) => {
+export const ConfigChangeListView = <T, >({ configChangeList, changeItemSummaryFunc }:
+                                      {configChangeList: ListChange<T, VersionedConfigChange>,
+                                      changeItemSummaryFunc: (item: T) => React.ReactNode}) => {
   if (!configChangeList.addedItems.length &&
-    !configChangeList.addedItems.length && !configChangeList.addedItems.length) {
+    !configChangeList.removedItems.length && !configChangeList.changedItems.length) {
     return <span className="fst-italic text-muted">no changes</span>
   }
   return <ul className="list-unstyled">
-    <li>Added: {configChangeList.addedItems.length}</li>
-    <li>Removed: {configChangeList.removedItems.length}</li>
-    <li>Changed: {configChangeList.changedItems.length}</li>
+    {configChangeList.addedItems.length > 0 && <li className="ps-4">Added: {configChangeList.addedItems.length}
+      <ul className="list-unstyled">
+        {configChangeList.addedItems.map((item, index) => <li className="ps-4" key={index}>
+          {changeItemSummaryFunc(item)}
+        </li>)}
+      </ul>
+    </li>}
+    {configChangeList.removedItems.length > 0 && <li className="ps-4">Removed: {configChangeList.removedItems.length}
+      <ul className="list-unstyled">
+        {configChangeList.removedItems.map((item, index) => <li className="ps-4" key={index}>
+          {changeItemSummaryFunc(item)}
+        </li>)}
+      </ul>
+    </li>}
+    {configChangeList.changedItems.length > 0 && <li className="ps-4">Changed: {configChangeList.changedItems.length}
+      <ul className="list-unstyled">
+        {configChangeList.changedItems.map((item, index) => <li className="ps-4" key={index}>
+          {versionedConfigChangeSummary(item)}
+        </li>)}
+      </ul>
+    </li>}
   </ul>
 }
+
+/**
+ *
+ */
```

Comment:
Needs JSDoc.

Reply:
ugh, yeah, I need to make sure I run 'lint' first, before lint --fix, because the latter silently puts all these empty comments in